### DISCUSSION TITLE
Store uploaded files data in Upload class objects?

### DISCUSSION THREAD

**Discussion by me21** [zauberzeug/nicegui#154] [Ideas / Feature Requests]
Ordinary components' values can be accessed as `value` attribute, whereas no similar attribute exists in Upload class.
Maybe store the uploaded files in `files` attribute?
I'm not sure about this, because on another hand it could cause memory consumption increase because reference to file contents will be around for longer than just in `on_upload` handler.
Consider this as an idea rather than an issue.

**Comment by gbrandt** [comment]
ui.upload has an upload complete event:

`self.upload_image = ui.upload(on_upload=self.__on_upload_complete)`

```
def __on_upload_complete( in_event ):

            raw_uploaded_bytes =in_event.files[0]
```

Is that what you are looking for?  It is done this way because multiple files can be uploaded with one `ui.upload()`

**Comment by me21** [comment]
Yes, I know about that handler. I just thought it could be nice to store uploaded files data in `files` attribute of ui.upload because of uniformity between Upload and Label/Input/Select/etc. - the value of the control is inside the object itself.

**Comment by falkoschindler** [comment]
I'd argue that the result of an event should not automatically be part of the UI element's state. You also wouldn't expect a button to have x/y properties containing the mouse coordinates of the last click event. These event arguments are rather volatile and usually only of value for those consumers that subscribe to the event in the first place.

The value of input elements is a bit different - in my opinion: The value is often part of the visible state, e.g. the content of an input field, the position of a slider or the state of a checkbox. Furthermore, it's (initial) value often can be set programmatically, making a r/w property very handy.

**Comment by me21** [comment]
Thank you for your input. That's why I made it a discussion rather than issue.
